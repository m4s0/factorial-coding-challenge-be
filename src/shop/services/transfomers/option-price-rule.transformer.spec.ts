import { OptionPriceRule } from '@Shop/entities/option-price-rule.entity';
import { transformOptionPriceRule } from '@Shop/services/transfomers/option-price-rule.transformer';
import { ProductOption } from '@Shop/entities/product-option.entity';

describe('transformOptionPriceRule', () => {
  it('should transform option price rule correctly', () => {
    const optionPriceRule = {
      id: 'fb5d0e88-8014-4883-82d5-94d0af8e723f',
      price: 29.99,
      targetOptionId: '4636127e-769e-4136-9a6a-8fddb7ca9692',
      targetOption: {
        id: '4636127e-769e-4136-9a6a-8fddb7ca9692',
        name: 'Color',
        displayName: 'Product Color',
        basePrice: 19.99,
        isActive: true,
        createdAt: new Date('2024-01-01'),
        updatedAt: new Date('2024-01-02'),
      } as ProductOption,
      dependentOptionId: '755cf8ae-950c-4be2-9a20-1fb6576b745e',
      dependentOption: {
        id: '3df0a970-d440-4361-940c-07a8ef39c333',
        name: 'Color',
        displayName: 'Product Color',
        basePrice: 19.99,
        isActive: true,
        createdAt: new Date('2024-01-01'),
        updatedAt: new Date('2024-01-02'),
      } as ProductOption,
      isActive: true,
      createdAt: new Date('2024-01-01'),
      updatedAt: new Date('2024-01-02'),
    } as OptionPriceRule;

    const result = transformOptionPriceRule(optionPriceRule);

    expect(result).toMatchSnapshot();
  });
});
