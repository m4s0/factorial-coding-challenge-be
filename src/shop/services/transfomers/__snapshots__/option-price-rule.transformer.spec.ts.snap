// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transformOptionPriceRule should transform option price rule correctly 1`] = `
{
  "createdAt": "2024-01-01T00:00:00.000Z",
  "dependentOption": {
    "basePrice": 19.99,
    "createdAt": "2024-01-01T00:00:00.000Z",
    "displayName": "Product Color",
    "id": "3df0a970-d440-4361-940c-07a8ef39c333",
    "isActive": true,
    "name": "Color",
    "updatedAt": "2024-01-02T00:00:00.000Z",
  },
  "dependentOptionId": "755cf8ae-950c-4be2-9a20-1fb6576b745e",
  "id": "fb5d0e88-8014-4883-82d5-94d0af8e723f",
  "isActive": true,
  "price": 29.99,
  "targetOption": {
    "basePrice": 19.99,
    "createdAt": "2024-01-01T00:00:00.000Z",
    "displayName": "Product Color",
    "id": "4636127e-769e-4136-9a6a-8fddb7ca9692",
    "isActive": true,
    "name": "Color",
    "updatedAt": "2024-01-02T00:00:00.000Z",
  },
  "targetOptionId": "4636127e-769e-4136-9a6a-8fddb7ca9692",
  "updatedAt": "2024-01-02T00:00:00.000Z",
}
`;
